### PR TITLE
Issue 190

### PR DESCRIPTION
**Pull request by JuergenKosel**
This pull request provides a fix for #190 

Commit 0bd845fec90fc1f2d77e74757f2e50db4f42bd4c fixes a race condition.
After applying this commit, issue 190 occured each time. Which allowed me to provide a fix for #190 in the 2 following commit.

### CODE REVIEW

**Comment by JuergenKosel** [comment]
The continuous integration check has failed, because the openssl header file is not in the include search path anymore. But this pull request does not touch the include search path nor any include directive.
Since also other contributers complained on such issues, I believe that some thing is wrong with the installation of the openssl library/headers on the continuous integration machine.

**Comment by icraggs** [comment]
Trying this PR in the fixes branch resulted in async client test failures:

https://travis-ci.org/eclipse/paho.mqtt.c/builds/219224597

I'm looking closer.

**Comment by JuergenKosel** [comment]
Hello @icraggs 
commit 45409c0063f34d864666ece0724c0cf8b0c698fc fixes a race condition.
Maybe this commit uncovers an issue, which is already there (e.g. issue #182 from the description and the failed unit tests).
Could you please verify this, by just merging / cherry picking this commit and rerun your test suite?

Greetings
Juergen

File: README.md [hunk 1]
```diff
@@ -127,6 +127,16 @@ python ../test/mqttsas2.py &
...
+```
+export TRAVIS_OS_NAME=linux
+export TRAVIS_BUILD_DIR=`pwd`
+PATH=$PATH:/usr/sbin ./travis-install.sh
+rm -Rf build.paho/
+travis-build.sh
+```
+
 
 ### Cross compilation
```
Comment (JuergenKosel) [inline README.md:141]: Is it still correct to report bugs to the eclipse bug tracker, as stated in line 209?

**Comment by JuergenKosel** [comment]
For me it looks like that some unit tests fail due to python exceptions, which are difficult to find (and fix).

**Comment by icraggs** [comment]
I added commit 45409c0 into the develop branch a week or so back.  Thanks for that.

**Comment by icraggs** [comment]
I'll update the README to say that bugs should be raised on Github.

**Comment by icraggs** [comment]
README updated.  Fix for 190 added to develop branch.  Thanks.

**Pull request closed without merging**